AR-Top-MobileApp/Assets/Prefabs/JSONReader.cs

```diff
@@ -0,0 +1,239 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+using Vuforia;
+
+public class JSONReader : MonoBehaviour {
+
+	//Simple prefabs that can be altered later.
+	[SerializeField] GameObject charPrefab;
+	[SerializeField] GameObject tilePrefab;
+	[SerializeField] GameObject gridPrefab;
+	[SerializeField] GameObject npcPrefab;
+
+	//GameObjects that will be used to organize the hierarchy and scale the layout.
+	private GameObject mapScaler;
+	private GameObject gridLayout;
+	private GameObject playerLayout;
+	private GameObject tileLayout;
+	private GameObject npcLayout;
+
+	private int gridHeightGap = 2; //The height gap for y coordinate grid spaces. Every increment represents 5 gamefoot.
+
+	//GridItem is serializable to make it JSON friendly.
+	[Serializable]
+	public class GridItem
+	{
+		public string type = null; //The type of the piece. Could be "character", "tile", or "npc".
```

Comment:
Shouldn't the type be something like _voxel, player, stone_wall stone_floor, etc._ Things that are specific to the type of tile being displayed.

Reply:
This is just the default value if no type is given through the editor.  It can default to whatever we want.